memgpt/server/rest_api/app.py

```diff
@@ -0,0 +1,169 @@
+import json
+import logging
+import secrets
+from pathlib import Path
+from typing import Optional
+
+import typer
+import uvicorn
+from fastapi import FastAPI
+from starlette.middleware.cors import CORSMiddleware
+
+from memgpt.server.constants import REST_DEFAULT_PORT
+
+# NOTE(charles): these are extra routes that are not part of v1 but we still need to mount to pass tests
+from memgpt.server.rest_api.auth.index import (
+    setup_auth_router,  # TODO: probably remove right?
+)
+from memgpt.server.rest_api.interface import StreamingServerInterface
+from memgpt.server.rest_api.routers.openai.assistants.assistants import (
+    router as openai_assistants_router,
+)
+from memgpt.server.rest_api.routers.openai.assistants.threads import (
+    router as openai_threads_router,
+)
+from memgpt.server.rest_api.routers.openai.chat_completions.chat_completions import (
+    router as openai_chat_completions_router,
+)
+
+# from memgpt.orm.utilities import get_db_session  # TODO(ethan) reenable once we merge ORM
+from memgpt.server.rest_api.routers.v1 import ROUTERS as v1_routes
+from memgpt.server.rest_api.routers.v1.users import (
+    router as users_router,  # TODO: decide on admin
+)
+from memgpt.server.rest_api.static_files import mount_static_files
+from memgpt.server.server import SyncServer
+from memgpt.settings import settings
+
+# TODO(ethan)
+# NOTE(charles): @ethan I had to add this to get the global as the bottom to work
+interface: StreamingServerInterface = StreamingServerInterface
```

Comment:
Note @norton120 I had to add these back since they weren't (afaik) getting initialized in global space anywhere else

Reply:
yep interfaces is a thing we need to dig into - I was under the impression that they were no longer used anywhere, yet they obviously are so that's a thing to tackle